### PR TITLE
[arci-ros2] Add arci-ros2 (only MoveBase trait)

### PR DESCRIPTION
**Pull request by OTL**
It requires ROS2 Foxy (Ubuntu 20.04).

* r2r is not published to crates.io yet.
* It requires ROS2 foxy is installed.

### CODE REVIEW

File: .github/workflows/coverage.yaml [hunk 1]
```diff
@@ -26,9 +26,11 @@ jobs:
       - run: ci/ubuntu-install-dependencies.sh
       - name: Generate code coverage
         run: |
+          # for test of arci-ros2
+          source /opt/ros/foxy/setup.bash
           # for test of arci-ros (roscore, rostopic)
-          source /opt/ros/melodic/setup.bash
-          cargo +nightly-2021-03-04 tarpaulin --verbose --all-features --workspace --out Xml
+          source /opt/ros/noetic/setup.bash
```
Comment (OTL) [inline .github/workflows/coverage.yaml:32]: coverage on ubuntu20.04 (noetic/foxy) only.
melodic is excluded.

File: arci-ros2/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+[dependencies]
+anyhow = "1.0"
+arci = "0.0.5"
+async-trait = "0.1"
+thiserror = "1.0"
+r2r = { git = "https://github.com/sequenceplanner/r2r", optional = true}
+
+[dev-dependencies]
+tokio = { version = "1.0", features = ["full"] }
```
Comment (OTL) [inline arci-ros2/Cargo.toml:21]: @taiki-e Is it OK to use full?

Comment (taiki-e) [reply arci-ros2/Cargo.toml:21]: Yes, it's fine. We can adjust it later if necessary. (Also, I guess the relevant C++ code or bindings probably won't work with WASM.)

File: arci-ros2/src/cmd_vel_move_base.rs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        twist_msg.angular.z = velocity.theta;
+        self.vel_publisher
+            .publish(&twist_msg)
+            .map_err(|e| arci::Error::Connection {
+                message: format!("r2r publish error: {:?}", e),
+            })
+    }
+
+    fn current_velocity(&self) -> Result<BaseVelocity, Error> {
+        panic!("not implemented yet");
```
Comment (mitsuharu-kojima) [inline arci-ros2/src/cmd_vel_move_base.rs:34]: panic!("not implemented yet") -> todo!() ?

File: arci-ros2/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+//! [`arci`] implementation using ROS1.
```
Comment (mitsuharu-kojima) [inline arci-ros2/src/lib.rs:1]: ROS1 -> ROS2 ?

File: arci-ros2/src/cmd_vel_move_base.rs (hunk 3 above)
Comment (OTL) [reply arci-ros2/src/cmd_vel_move_base.rs:34]: I googled and replaced with unimplemented.

File: arci-ros2/src/lib.rs (hunk 4 above)
Comment (OTL) [reply arci-ros2/src/lib.rs:1]: Done

**Comment by OTL** [comment]
r2r is broken now. I'm working on it.
https://github.com/sequenceplanner/r2r/pull/10

**Comment by OTL** [comment]
r2r is fixed. They are using Rolling (the latest (unstable?) distro for ROS2) instead of foxy.

**Review by mitsuharu-kojima — APPROVED** [review]

**Review by taiki-e — APPROVED** [review]

**Pull request merged**